cli/src/pcluster/validators/s3_validators.py

```diff
@@ -38,21 +51,26 @@ def _validate(self, url, retries=3, fail_on_https_error: bool = False, fail_on_s
                 FailureLevel.ERROR,
             )
 
-    def _validate_s3_uri(self, url: str, fail_on_error: bool):
+    def _validate_s3_uri(self, url: str, fail_on_error: bool, expected_bucket_owner: str):
         try:
             match = re.match(r"s3://(.*?)/(.*)", url)
             if not match or len(match.groups()) < 2:
                 self._add_failure(f"s3 url '{url}' is invalid.", FailureLevel.ERROR)
             else:
                 bucket_name, object_name = match.group(1), match.group(2)
-                AWSApi.instance().s3.head_object(bucket_name=bucket_name, object_name=object_name)
+                AWSApi.instance().s3.head_object(
+                    bucket_name=bucket_name, object_name=object_name, expected_bucket_owner=expected_bucket_owner
+                )
 
-        except AWSClientError:
+        except AWSClientError as e:
             # Todo: Check that bucket is in s3_read_resource or s3_read_write_resource.
-            self._add_failure(
-                f"The S3 object '{url}' does not exist or you do not have access to it.",
-                FailureLevel.ERROR if fail_on_error else FailureLevel.WARNING,
-            )
+            if expected_bucket_owner:
+                self._add_failure(str(e), FailureLevel.ERROR)
+            else:
+                self._add_failure(
+                    f"The S3 object '{url}' does not exist or you do not have access to it.",
+                    FailureLevel.ERROR if fail_on_error else FailureLevel.WARNING,
+                )
```

Comment:
can we make the messaging homogeneous between these 2 cases? For example by always using the message from the exception and improve it if needed?

Reply:
would it make sense to add a mention to the url in the exception message? I'm wondering in what case of the 3 you mentioned we fall in case the s3 bucket exists but the object does not.

Reply:
The s3 bucket exist but object does not failed with the third case. "Unexpected error when getting S3 bucket '{bucket_name}': '{error_message}'"
The error message is:
The S3 object '{url}' does not exist or you do not have access to it."
Now adding 
```
    elif object_name and error_code == "404" and error_message == "Not Found":
        message = f"Failed when accessing object '{object_name}' from bucket '{bucket_name}'. This can be due to " \
                  f"'{object_name}' not found in '{bucket_name}'"
```
when processing `_process_s3_bucket_error`to recognize this case, the error message become:
"Failed when accessing object 'scheduler_plugin/artifact1' from bucket 'amibucket11223'. This can be due to 'scheduler_plugin/artifact1' not found in 'amibucket11223'"